src/test/java/de/tum/cit/aet/artemis/buildagent/service/TestResultXmlParserTest.java

```diff
@@ -240,10 +240,77 @@ void testNestedTestsuite() throws IOException {
                 """;
 
         TestResultXmlParser.processTestResultFile(input, failedTests, successfulTests);
-        assertThat(successfulTests).hasSize(12).extracting(BuildResult.LocalCITestJobDTO::getName).containsExactlyInAnyOrder("Testing filtering in A", "Testing mapping in A",
-                "Testing filtering in B", "Testing mapping in B", "Testing filtering in C", "Testing mapping in C", "Testing A against sample solution",
-                "Testing B against sample solution", "Testing C against sample solution", "Testing selectAndReflectA (0,0) []", "Testing selectAndReflectB (0,1) [(0,0)]",
-                "Testing selectAndReflectC (0,1) [(-1,-1)]");
+
+        // @formatter:off
+        assertThat(successfulTests).extracting(BuildResult.LocalCITestJobDTO::getName).containsExactlyInAnyOrder(
+            "Properties.Checked by SmallCheck.Testing filtering in A",
+                "Properties.Checked by SmallCheck.Testing mapping in A",
+                "Properties.Checked by SmallCheck.Testing filtering in B",
+                "Properties.Checked by SmallCheck.Testing mapping in B",
+                "Properties.Checked by SmallCheck.Testing filtering in C",
+                "Properties.Checked by SmallCheck.Testing mapping in C",
+                "Properties.Checked by QuickCheck.Testing A against sample solution",
+                "Properties.Checked by QuickCheck.Testing B against sample solution",
+                "Properties.Checked by QuickCheck.Testing C against sample solution",
+                "Unit Tests.Testing selectAndReflectA (0,0) []",
+                "Unit Tests.Testing selectAndReflectB (0,1) [(0,0)]",
+                "Unit Tests.Testing selectAndReflectC (0,1) [(-1,-1)]");
+        // @formatter:on
+        assertThat(failedTests).isEmpty();
+    }
+
+    @Test
+    void testMultipleTestsuite() throws IOException {
```

Comment:
can you add a few more examples with different nesting structures?

Reply:
More tests added in https://github.com/ls1intum/Artemis/pull/9691/commits/cda222a86afc911b8f139ba888e92aed811adf3d